### ISSUE TITLE
Improve App Sign In / Sign Out

### ISSUE THREAD

**Issue by vusters** [EnsembleUI/ensemble#1101]

**Comment by kmahmood74** [comment]
@vusters no, no one would do this. You don't put screens behind auth, you put data behind it. When you go from one screen to another, you make an API call and that should return a 401 unauthorized error that you should use to redirect the user.

**Comment by amin-nas** [comment]
Not exactly a 1-1 comparison - In React, you can handle navigation in one place and put routes (screens in our case) behind authentication.
What is the point of forcing the app to load a screen, make an API call, wait for response, and then route to login if the app already knows user is not logged in.

**Comment by kmahmood74** [comment]
because the app doesn't know if the user is logged in or not without going to the server :-)

**Comment by amin-nas** [comment]
We could have a global way of checking auth, and then specify which screens to not load if not authed.
The goal is to avoid repeating the same logic on each individual screen. As long as we can achieve that, app devs would be happy!

**Comment by kmahmood74** [comment]
yes - I think the way to do is to specify at the app level what to do when API returns a specific header or content. The HTTP header for unauthorized access is 401 but some apps send errors in the body with a 200 error. Let the server return it instead of proactively checking.

**Comment by vusters** [comment]
To be clear, this is strictly a client-side best effort to verify sign in status. HTTP status is a different effort.
